Release notes for Foreman 2.4.0 (#1794)
theforeman/theforeman.org: _includes/manuals/2.4/1.2_release_notes.md

### Headline features

#### Ansible inventory template

> We've already eliminated the necessity for Katello to use Ansible Inventory[31193](https://projects.theforeman.org/issues/31193). This change eliminates the need for the Ansible plugin to render Ansible inventory. ([31482](https://projects.theforeman.org/issues/31482))

Reviewer: ```suggestion
We've already eliminated the necessity for Katello to use Ansible Inventory ([#31193](https://projects.theforeman.org/issues/31193)). This change eliminates the need for the Ansible plugin to render Ansible inventory. ([#31482](https://projects.theforeman.org/issues/31482))
```

See the upgrade warnings for more information.

####  Personal Access Tokens

> The API endpoints were introduced for Personal Access Tokens in Foreman 1.17 version. Now WebUI can be used to manage Personal Access Tokens. A new token can be generated by the user and set to expire at a certain time and date. To make tokens void before their expiration date and time, one can use the revoke action. ([31080](https://projects.theforeman.org/issues/31080))

Reviewer: ```suggestion
Personal Access Tokens were introduced in Foreman 1.17 version with API endpoints only. Now WebUI can be used to manage Personal Access Tokens. A new token can be generated by the user and set to expire at a certain time and date. To make tokens void before their expiration date and time, one can use the revoke action. ([#31080](https://projects.theforeman.org/issues/31080))
```

> Personal Access Tokens were introduced in Foreman 1.17 version with API endpoints only. Now WebUI can be used to manage Personal Access Tokens. A new token can be generated by the user and set to expire at a certain time and date. To make tokens void before their expiration date and time, one can use the revoke action. ([#31080](https://projects.theforeman.org/issues/31080))

Reviewer 2: ```suggestion
Personal Access Tokens were introduced in Foreman 1.17 version with API endpoints only. Now, you can manage Personal Access Tokens through the Foreman web UI. You can generate a new token and set it to expire at a certain date and time. To make tokens void before their expiration date and time, you can use the revoke action. ([#31080](https://projects.theforeman.org/issues/31080))
```

* New tab: Hover to current logged in user -> _My Account_ -> _Personal Access Tokens_

#### Support for Redfish to BMC Smart Proxy

> The one more BMC provider Redfish is now available. Some of the features for example, boot media aren't available across all vendors (HPE and Dell in particular). The Redfish provider for the BMC Smart Proxy solves this issue while also potentially allowing additional BMC functions to be integrated. ([29636](https://projects.theforeman.org/issues/29636))

Reviewer: ```suggestion
Redish is now available as another BMC provider. Some of the features, such as boot media, weren't available across all vendors (HPE and Dell in particular). The Redfish provider for the BMC Smart Proxy solves this issue while also potentially allowing additional BMC functions to be integrated. ([#29636](https://projects.theforeman.org/issues/29636))
```

See the upgrade warnings for more information.

####  Personal Access Tokens

> WebUI can be used to handle Personal Access Tokens. A new token can be generated by the user and set to expire at a certain time and date.

Reviewer: Maybe it's good to clarify that previously it only had an API.

To make tokens void before their expiration date and time, one can use the revoke action. ([31080](https://projects.theforeman.org/issues/31080))

* New tab: Hover to current logged in user -> _My Account_ -> _Personal Access Tokens_

#### Support for Redfish to BMC Smart Proxy

> The Intelligent Platform Management Interface (IPMI) is being phased out in favour of Redfish for network hardware management. Some of the features for example, boot media aren't available across all vendors (HPE and Dell in particular). The Redfish provider for the BMC Smart Proxy solves these issues while also potentially allowing additional BMC functions to be integrated. ([29636](https://projects.theforeman.org/issues/29636))

Reviewer 3: @lzap can you take a look at this one please?

Reviewer: Are there plans to phase out IPMI? Not all hardware supports Redfish so I think phasing out is too strong.

Reviewer 4: No there are no plans at the moment, Redfish was just contributed by one of our community users and it is very basic. We currently have not plans on giving up IPMI.

Basically just say that Redfish is now available as another BMC provider.

* Clean up deprecations for 2.4 ([#31233](https://projects.theforeman.org/issues/31233))
* Add Hostgroup API views Facet extenstion points ([#31213](https://projects.theforeman.org/issues/31213))

#### Foreman - Proxy gateway
* Repository enable failed on http-proxy configured setup. ([#31637](https://projects.theforeman.org/issues/31637))

#### Foreman - Puppet Reports
* WebUI spinner spins forever when clicking a column within config report ([#31526](https://projects.theforeman.org/issues/31526))

#### Foreman - Puppet integration
* The fix for #31168 overwrites $::hostname with the FQDN in Puppet ([#31882](https://projects.theforeman.org/issues/31882))
* Puppet Environments are not listed properly in the SCP details page ([#31625](https://projects.theforeman.org/issues/31625))
* Improve /puppetrun error message ([#31543](https://projects.theforeman.org/issues/31543))
* Each smart class parameter appears twice, when having two environments ([#31538](https://projects.theforeman.org/issues/31538))
* TemplateInput type registry ([#31284](https://projects.theforeman.org/issues/31284))

> * Add warning banner about puppet ENC extraction ([#30215](https://projects.theforeman.org/issues/30215))

Reviewer: I think this deserves a section in deprecations.

* update boostrap script to configure subscription-manager to use /pulp/content/ instead of /pulp/repos/ ([#31868](https://projects.theforeman.org/issues/31868))
* Create qdrouter inbound link for katello.agent queue ([#31800](https://projects.theforeman.org/issues/31800))
* Create pulp.task queue for katello agent ([#31761](https://projects.theforeman.org/issues/31761))
* Remove foreman-proxy-plugin-pulp from Foreman scenario ([#31757](https://projects.theforeman.org/issues/31757))
* Restore katello::qpid_client ([#31693](https://projects.theforeman.org/issues/31693))
* Command exceeded timeout while Installer executes foreman-rake db:migrate ([#31670](https://projects.theforeman.org/issues/31670))
* Foreman content proxy doesn't listen on port 80 anymore ([#31662](https://projects.theforeman.org/issues/31662))
* The container gateway smart proxy plugin needs installer support ([#31642](https://projects.theforeman.org/issues/31642))
* Deploy Pulp 3 by default on Foreman server and as a mirror configuration on content proxies ([#31614](https://projects.theforeman.org/issues/31614))

> * foreman-proxy-certs-generate fails with custom certificates ([#31611](https://projects.theforeman.org/issues/31611))

Reviewer: I changed this category

* Increase Artemis disk usage limit ([#31607](https://projects.theforeman.org/issues/31607))
* Allow changing some Foreman Proxy and Apache hostnames on Katello-based certificates ([#31509](https://projects.theforeman.org/issues/31509))
* Update candlepin broker.xml for jobs processing ([#31497](https://projects.theforeman.org/issues/31497))
* Remove digital ocean plugin ([#31490](https://projects.theforeman.org/issues/31490))

> * Upgrade results in empty /var/opt/rh/rh-postgresql12/lib/pgsql/data which causes upgrade to fail ([#31486](https://projects.theforeman.org/issues/31486))

Reviewer: I changed this category

* Ensure import/export directories are created and have proper permissions ([#31468](https://projects.theforeman.org/issues/31468))
* Drop ssl_protocol parameter on foreman_proxy_content ([#31435](https://projects.theforeman.org/issues/31435))
* Installer can't install the Ansible plugin on Debian/Ubuntu ([#31430](https://projects.theforeman.org/issues/31430))
* Unable to set ping_free_ip in Foreman Proxy's dhcp.yml ([#31415](https://projects.theforeman.org/issues/31415))

> * Disable TLS 1.0 and 1.1 by default in Apache ([#31387](https://projects.theforeman.org/issues/31387))

Reviewer: I changed this category
